app/src/main/java/fr/free/nrw/commons/depictions/SubClass/SubDepictionListFragment.java

```diff
@@ -0,0 +1,178 @@
+            handleNoInternet();
+        } else {
+            progressBar.setVisibility(View.VISIBLE);
+            try {
+                presenter.initSubDepictionList(qid, isParentClass);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_browse_image, container, false);
+        ButterKnife.bind(this, v);
+        presenter.onAttachView(this);
+        isParentClass = false;
+        depictionNotFound.setVisibility(GONE);
+        if (getActivity().getResources().getConfiguration().orientation
+                == Configuration.ORIENTATION_PORTRAIT) {
+            layoutManager = new LinearLayoutManager(getContext());
+        } else {
+            layoutManager = new GridLayoutManager(getContext(), 2);
+        }
+        initViews();
+        depictionsRecyclerView.setLayoutManager(layoutManager);
+        ArrayList<DepictedItem> items = new ArrayList<>();
+        depictionsAdapter = adapterFactory.create(items);
+        depictionsRecyclerView.setAdapter(depictionsAdapter);
+        return v;
+    }
+
+    private void handleNoInternet() {
+        progressBar.setVisibility(GONE);
+        ViewUtil.showShortSnackbar(depictionsRecyclerView, R.string.no_internet);
+    }
+
+    @Override
+    public void onImageUrlFetched(String response, int position) {
+        depictionsAdapter.getItem(position).setImageUrl(response);
+        depictionsAdapter.notifyItemChanged(position);
+    }
+
+    @Override
+    public void onSuccess(List<DepictedItem> mediaList) {
+        isLoading = false;
+        hasMoreImages = false;
+        progressBar.setVisibility(View.GONE);
+        depictionNotFound.setVisibility(GONE);
+        bottomProgressBar.setVisibility(GONE);
+        int itemCount=layoutManager.getItemCount();
+        depictionsAdapter.addAll(mediaList);
+        if(itemCount!=0) {
+            depictionsAdapter.notifyItemRangeInserted(itemCount, mediaList.size()-1);
+        }else{
+            depictionsAdapter.notifyDataSetChanged();
+        }
+    }
+
+    @Override
+    public void initErrorView() {
+        hasMoreImages = false;
+        progressBar.setVisibility(GONE);
+        bottomProgressBar.setVisibility(GONE);
+        depictionNotFound.setVisibility(VISIBLE);
+        String no_depiction = getString(isParentClass? R.string.no_parent_classes: R.string.no_child_classes);
+        depictionNotFound.setText(String.format(Locale.getDefault(), no_depiction, depictsName));
+
+    }
+
+    @Override
+    public void showSnackbar() {
+        ViewUtil.showShortSnackbar(depictionsRecyclerView, R.string.error_loading_depictions);
+    }
+
+    @Override
+    public void setIsLastPage(boolean b) {
```

Comment:
Is this not needed anymore?

Reply:
the code inside the function is missing. I'll add it